cc/modules/protocol/mpc/snn/src/impl/mpc_basic.cpp

```diff
@@ -81,9 +82,9 @@ int ComputeMSB::funcComputeMSB3PC(const vector<mpc_t>& a, vector<mpc_t>& b, size
     populateRandomVector<mpc_t>(r2, size, "a_2", "POSITIVE");
     for (size_t i = 0; i < size; ++i) {
       if (r1[i] == MINUS_ONE)
-        r1[i]--;
+        r1[i] = 0;////[huanggaofeng] fix for odd ring Z_{2^128 - 1}, origin: r1[i]--
```

Comment:
Could you explain why this works?

Reply:
In odd ring `Z_{L -1}`,  -1 means 0, which is -1 in `Z_L`